python/cudf/cudf/core/column/string.py

```diff
@@ -5918,7 +5918,7 @@ def values(self) -> cupy.ndarray:
         """
         Return a CuPy representation of the StringColumn.
         """
-        raise TypeError("String Arrays is not yet implemented in cudf")
+        raise TypeError("String Arrays is not yet implemented in cupy")
```

Comment:
Perhaps this is a question for other reviewers. Should this be a `NotImplementedError`?

Reply:
I would say that is more correct, yes

Reply:
We have to change this back because dask is expecting a TypeError from us, not a NotImplementedError (see https://github.com/dask/dask/blob/main/dask/dataframe/partitionquantiles.py#L452).

More generically I usually think of a NotImplementedError as something you raise for functionality that is "not _yet_ implemented", not something that will never be implemented, and AFAICT cupy plans to treat string arrays as out of scope unless they can eventually leverage another library (such as pylibcudf) for the implementation (i.e. they have no intention of ever implementing themselves).

Reply:
Okay. I'm going to apply Vyas's suggestions to save @dagardner-nv from the churn I caused here.

Reply:
Thanks Bradley.